source/Nevermore/Advanced/ReadTransaction.cs

```diff
@@ -905,22 +905,27 @@ DbTransaction BeginTransaction(IsolationLevel isolationLevel, string sqlServerTr
 
         public void Dispose()
         {
-            // ReSharper disable ConstantConditionalAccessQualifier
-            if (OwnsSqlTransaction)
+            try
             {
-                Transaction?.Dispose();
-            }
-
-            TransactionTimer?.Dispose();
-            DeadlockAwareLock?.Dispose();
+                // ReSharper disable ConstantConditionalAccessQualifier
+                if (OwnsSqlTransaction)
+                {
+                    Transaction?.Dispose();
+                }
 
-            if (OwnsSqlTransaction)
-            {
-                connection?.Dispose();
+                TransactionTimer?.Dispose();
+                DeadlockAwareLock?.Dispose();
             }
+            finally
+            {
+                if (OwnsSqlTransaction)
```

Comment:
My initial thought was: Is it safe to move this until after the TransactionTimer and DeadlockAwareLock are disposed?

I did a small dig

- TransactionTimer ends up just writing to the log. That's safe wherever you put it
   - It may now be slightly less accurate; if the `connection.Dispose` takes a non-zero amount of time, that used to be captured by the Timer and now won't be. I don't think this is worth worrying about.
   
- `DeadlockAwareLock` is just a `SemaphoreSlim`, also safe to dispose in either the old or new positions.

I'm happy there, approving the PR